packages/sdk/src/modules/did/did.spec.ts

```diff
@@ -94,18 +134,290 @@ describe('Did', () => {
+  const read_did = await did.read({
+    name: new_did
+  });
+
+  // // helpful debugging
+  // console.log("read_did", read_did, "\n for new name: ", new_did);
+  // console.log("did document", read_did?.document, "\n for new name: ", new_did);
+
+  // test name of set did
+  expect(read_did).toBeDefined();
+
+  // test specific did document information
+  await readDid(read_did as ReadDidResponse, new_did, alice, customFields || null);
+
+  // remove did for cleanup
+  const removeResult = await did.remove({name: new_did});
+  expect(removeResult?.hash).toBeDefined();
+  expect(typeof removeResult?.unsubscribe).toBe('function');
+
+}
+
+async function readDid(read_did: ReadDidResponse, new_did: string, alice: KeyringPair, customFields: CustomDocumentFields | null) {
+  expect(read_did).toBeDefined();
+  expect(read_did?.name).toBe(new_did);
+
+  // check value, validity, and created to be defined
+  expect(read_did?.value).toBeDefined();
+  expect(read_did?.validity).toBeDefined();
+  expect(read_did?.value).toBeDefined();
+
+  // value names based on expected regex formats
+  const valuePattern = /^0x[0-9a-fA-F]+$/;
+  const validityPattern = /^\d{1,3}(,\d{3})*$/;
+  const createdPattern = /^\d{1,3}(,\d{3})*$/;
+  
+  // makes sure all values align with expected regex
+  expect(valuePattern.test(read_did?.value as string)).toBe(true);
+  expect(validityPattern.test(read_did?.validity as string)).toBe(true);
+  expect(createdPattern.test(read_did?.created as string)).toBe(true);
+
+  // test common did document default values
+  expect(read_did?.document).toBeDefined();
+  const didDocument = read_did?.document;
+  expect(didDocument?.id).toBe(`did:peaq:${alice.address}`);
+  expect(didDocument?.controller).toBe(`did:peaq:${alice.address}`);
+
+  if (customFields == null){ // test did document default empty values
+    expect(didDocument?.verificationmethodsList).toEqual([]);
+    expect(didDocument?.signature).toBeUndefined(); // have undefined here, do we want empty list like others?
+    expect(didDocument?.servicesList).toEqual([]);
+    expect(didDocument?.authenticationsList).toEqual([]);
+  }
+  else { // test custom values if not default by looping thru the read did document to check against the expected set customFields
+
+    // Iterate through verifications
+    if (didDocument.verificationmethodsList) {
+      didDocument.verificationmethodsList.forEach(verification => {
+        customFields.verifications?.forEach(setVerification => {
+          const uuidPattern = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
+          expect(uuidPattern.test(verification.id as string)).toBe(true);
+          expect(verification.type).toEqual(setVerification.getType());
+          expect(verification.controller).toEqual(`did:peaq:${alice.address}`);
+          
+          // error here with proto doc
+          // expect(verification.publicKeyMultibase).toEqual(`z${alice.address}`);
```

Comment:
Currently having an issue where the proto document is not built correctly to test this line. You can see [here](https://github.com/peaqnetwork/peaq-network-did-proto-format/blob/feature/1206361464402337_develop-data-verification-flutter-peaq-sdk-package/javascript/did_document_format_pb.js) that the publickeymulibase does not use camelCase. The update [branch](https://github.com/peaqnetwork/peaq-network-did-proto-format/blob/feature/1207515694088897_proto_name_fix/javascript/did_document_format_pb.js) fixes this issue.